python/private/common/py_executable.bzl

```diff
@@ -91,6 +92,17 @@ filename in `srcs`, `main` must be specified.
     allow_none = True,
 )
 
+def is_target_platform_windows(ctx):
+    """Determine if the target platform is a windows platform.
+
+    Args:
+        ctx (ctx): The rule's context object
+
+    Returns:
+        bool: True if the target platform is Windows.
+    """
+    return ctx.target_platform_has_constraint(Label("{}/os:windows".format(PLATFORMS_LOCATION)))
```

Comment:
This works? I thought they had to be attr.label() dependencies.

I think this is incomplete, too. IIRC, there are multiple windows OS targets for the different variants (32, 64, etc). All of them should be listed and checked (just as done for the "is darwin" logic).

Reply:
Does passing CI not demonstrate this works? I made these changes mostly without running locally as I have no windows machine. Is this code path tested on all platforms in PR builds?

Reply:
CI only runs Ubuntu with the Starlark rules, and only a subset of tests (none of the integration tests or examples). Windows + the Starlark impl is actually broken today; I ran into it while fixing another bug. Once https://github.com/bazelbuild/bazel/commit/899e928bf2af79a60e4c6fd2595dae2b90cb02ed is available in last_green (might be already by now), I can merge in #1443 and then cleanup #1452 (which removes the get_current_os_name() calls and also adds tests for Windows).

And yes, I can with certainty say that passing a label is incorrect: target_platfrom_has_constraint expects a ConstraintValueInfo object.